### DISCUSSION TITLE
KMemo: How to center a table

### DISCUSSION THREAD

**Discussion by radekjindra** [kryslt/KControls#57] [Q&A] [answered]
Hello, I have a request. How do I center this table overall? Thank you very much for your help

{grey part of the table -->> header}
 TBL01_BP:=Protokol_ZKP.Blocks.AddTable;
 TBL01_BP.CellStyle.ContentPadding.AssignFromValues(5,5,5,5);
 TBL01_BP.CellStyle.ContentMargin.AssignFromValues(1,1,1,1);
 TBL01_BP.ApplyDefaultCellStyle;
 {columns}
 TBL01_BP.ColCount := 3;
 {lines}
 TBL01_BP.RowCount:=1;
 {line weight}
 TBL01_BP.CellStyle.BorderWidth := 2;
 TBL01_BP.CellStyle.BorderColor :=clBlack;
 {width of the column}
 TBL01_BP.ColWidths[0]:=100;
 TBL01_BP.ColWidths[1]:=200;
 TBL01_BP.ColWidths[2]:=80;
 {1 column}
 TBL01_BP.Cells[0, 0].Blocks.AddTextBlock('Measuring Point').TextStyle.Font.Style:=[fsBold];
 TBL01_BP.Cells[0, 0].ParaStyle.HAlign:=halCenter;
 TBL01_BP.Cells[0, 0].ParaStyle.BorderColor:=clBlack;
 TBL01_BP.ApplyDefaultCellStyle;
 {2 column}
 TBL01_BP.Cells[1, 0].Blocks.AddTextBlock('Soil Resistivity').TextStyle.Font.Style:=[fsBold];
 TBL01_BP.Cells[1, 0].ParaStyle.HAlign:=halCenter;
 TBL01_BP.Cells[1, 0].ParaStyle.BorderColor:=clBlack;
 TBL01_BP.ApplyDefaultCellStyle;
 {3 column}
 TBL01_BP.Cells[2, 0].Blocks.AddTextBlock('Level of Aggression').TextStyle.Font.Style:=[fsBold];
 TBL01_BP.Cells[2, 0].ParaStyle.HAlign:=halCenter;
 TBL01_BP.Cells[2, 0].ParaStyle.BorderColor:=clBlack;
 TBL01_BP.ApplyDefaultCellStyle;
 {end of table}
 TBL01_BP.CellStyle.Brush.Color :=$C0C0C0;
 TBL01_BP.ApplyDefaultCellStyle;

 {this is wrong i'm a beginner sorry}
 {center table }
 PA:=Protokol_ZKP.Blocks.AddParagraph();
 PA.ParaStyle.HAlign:=halCenter;

**Comment by kryslt** [comment, marked as answer]
```
procedure TMainForm.TestXX;
var
  TBL: TKMemoTable;
  PA: TKMemoParagraph;
begin
  // add table with a fixed width
  // the table is just another block, the fixed width of the table is necessary so that the following paragraph can center the block 
  TBL := KMemo1.Blocks.AddTable;
  TBL.FixedWidth := True;
  TBL.RequiredWidth := 600;
  // add horizontally centered paragraph
  PA := KMemo1.Blocks.AddParagraph;
  PA.ParaStyle.HAlign := halCenter;
end;
```
